jscomp/core/js_implementation.ml

```diff
@@ -164,9 +164,9 @@ let after_parsing_impl ppf outputprefix (ast : Parsetree.structure) =
         let lambda, exports =
           Translmod.transl_implementation modulename typedtree_coercion
         in
-        let js_program =
+        let js_program ~module_system =
           print_if_pipe ppf Clflags.dump_rawlambda Printlambda.lambda lambda
-          |> Lam_compile_main.compile outputprefix exports
+          |> Lam_compile_main.compile outputprefix module_system exports
```

Comment:
Here is the root cause breaking the tests

Reply:
It seems delaying the evaluation of js_program causing the test failure.

Reply:
See how it goes https://github.com/rescript-lang/rescript-compiler/pull/6191

Reply:
Bingo.